src/backend/routers/post.py

```diff
@@ -179,6 +184,25 @@ async def get_comments(
     )
 
 
+@router.post("/{post_id}/rating")
+async def create_rating(
+    user_id: Annotated[UUID, Depends(is_authorized)], post_id: str, rating: PostRatingBody
+) -> PostRating:
+    """Create a rating on the specified post."""
+    try:
+        rating_record = await PostRating.prisma().upsert(
+            where={"post_id_author_id": {"post_id": post_id, "author_id": str(user_id)}},
+            data={
+                "create": {"post_id": post_id, "author_id": str(user_id), "value": rating.rating},
+                "update": {"value": rating.rating},
+            },
+        )
+    except PrismaError as e:
+        logger.warning(f"Could not create rating: {e}")
+        raise HTTPException(400, "Could not create the rating due to an internal error")
```

Comment:
Internal errors should be 500, ya.
If you can think of any specific prisma errors, raise  4xx error codes in those cases and throw 5xx in the general case.